js/store.js

```diff
@@ -39,19 +40,22 @@ export class TodoStore extends EventTarget {
 					todos[i] = await state.get(todo['@id']);
 			}));
 		}
-		const config = {
-			'@id': uuid(),
-			'@domain': `${this.id}.public.gw.m-ld.org`,
-			genesis: isNew,
-			io: {uri: `https://gw.m-ld.org`}
-		};
+		const device = await Device.here();
+		// Exchange the access token for gateway domain configuration
+		const configRes = await fetch(`/api/config?domain=${this.id}`, {
+			method: 'POST',
+			headers: {'Authorization': `Bearer ${token}`, 'Content-Type': 'application/json'},
+			body: JSON.stringify(device.key)
+		});
+		const {config, pid} = await configRes.json();
 		const meld = await clone(
 			new MemoryLevel,
 			IoRemotes,
-			config
+			{'@id': uuid(), ...config},
+			device.asMeldApp(pid)
 		);
```

Comment:
Here we replace the hard-wired clone configuration with the configuration received from the lambda. Ne special handling is needed for the Gateway JWT embedded in the configuration, we just pass it all to **m-ld**, remembering to add the `@id` for the clone. We also provide the signatures implementation to **m-ld** via the "device".